extensions/functions_datetime.yaml

```diff
@@ -1,36 +1,52 @@
 %YAML 1.2
 ---
 scalar_functions:
-  -
-    name: extract
+  - name: extract
     description: Extract portion of a date/time value.
     impls:
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, DAY_OF_YEAR, ISO_WEEK, IS_LEAP_YEAR, HOUR, SECOND, SUBSECOND, MILLISECOND, MICROSECOND, NANOSECOND ]
             name: The part of the value to extract.
             required: true
           - value: timestamp
         return: i64
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ IS_LEAP_YEAR ]
+            name: The part of the value to extract.
+            required: true
+          - value: timestamp
+        return: boolean
+      - args:
+          - options: [ YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, DAY_OF_YEAR, ISO_WEEK, IS_LEAP_YEAR, HOUR, SECOND, SUBSECOND, MILLISECOND, MICROSECOND, NANOSECOND, IS_DST ]
```

Comment:
Can you add definitions and what common data tools use each to a comment somewhere? That will make it easier to evaluate whether things should be included.